docs/01.CONFIGURATION.md

```diff
@@ -0,0 +1,249 @@
+  | C:      | `-DJERRY_LOGGING=0/1`                        |
+  | CMake:  | `-DJERRY_LOGGING=ON/OFF`                     |
+  | Python: | `--logging=ON/OFF`                           |
+
+* ### LCache
+
+  This option enables the LCache, allowing faster access to object properties. See [Internals](04.INTERNALS.md) for further details.
+  This option is enabled by default.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_LCACHE=0/1`                         |
+  | CMake:  | `<none>`                                     |
+  | Python: | `<none>`                                     |
+
+* ### Property hashmaps
+
+  This option enables the creation of hashmaps for object properties. See [Internals](04.INTERNALS.md) for further details.
+  This option is enabled by default.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_PROPRETY_HASHMAP=0/1`               |
+  | CMake:  | `<none>`                                     |
+  | Python: | `<none>`                                     |
+
+* ### Memory statistics
+
+  This option can be used to provide memory usage statistics either upon engine termination, or during runtime using the `jerry_get_memory_stats` jerry API function.
+  The feature can create a significant performance overhead, and should only be used for measurement purposes. This option is disabled by default.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_MEM_STATS=0/1`                      |
+  | CMake:  | `-DJERRY_MEM_STATS=ON/OFF`                   |
+  | Python: | `--mem-stats=ON/OFF`                         |
+
+* ### Heap size
+
+  This option can be used to adjust the size of the internal heap, represented in kilobytes. The provided value should be an integer, with a maximum value of 512.
+  The default value is 512.
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_GLOBAL_HEAP_SIZE=(int)`             |
+  | CMake:  | `--DJERRY_GLOBAL_HEAP_SIZE=(int)`            |
+  | Python: | `--heap-size=(int)`                          |
+
+* ### Stack limit
+
+  This option can be used to cap the stack usage of the engine, and prevent stack overflows due to recursion. The provided value should be an integer, which represents the allowed stack usage in kilobytes.
+  The default value is 0 (unlimited).
+
+  | Options |                                              |
+  |---------|----------------------------------------------|
+  | C:      | `-DJERRY_STACK_LIMIT=(int)`                  |
+  | CMake:  | `-DJERRY_STACK_LIMIT=(int)`                  |
+  | Python: | `--stack-limit=(int)`                        |
```

Comment:
This patch (#2969) haven't been landed yet.

Reply:
I added this ahead of time, so that we won't have any dependencies between the PR's.